examples/tic-tac-toe/ui/src/hooks/useTransactions.ts

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { bcs } from '@mysten/sui/bcs';
+import { PublicKey } from '@mysten/sui/cryptography';
+import { ObjectRef, Transaction } from '@mysten/sui/transactions';
+import { useNetworkVariable } from 'config';
+import { Game } from 'hooks/useGameQuery';
+import { TurnCap } from 'hooks/useTurnCapQuery';
+import { multiSigPublicKey } from 'MultiSig';
+
+/** Hook to provide an instance of the Transactions builder. */
+export function useTransactions(): Transactions | null {
+	const packageId = useNetworkVariable('packageId');
+	return packageId ? new Transactions(packageId) : null;
+}
+
+/**
+ * Builds on-chain transactions for the Tic-Tac-Toe game.
+ */
+export class Transactions {
+	readonly packageId: string;
+
+	constructor(packageId: string) {
+		this.packageId = packageId;
+	}
+
+	newSharedGame(player: string, opponent: string): Transaction {
+		const tx = new Transaction();
+
+		tx.moveCall({
+			target: `${this.packageId}::shared::new`,
+			arguments: [tx.pure.address(player), tx.pure.address(opponent)],
+		});
+
+		return tx;
+	}
+
+	newMultiSigGame(player: PublicKey, opponent: PublicKey): Transaction {
+		const admin = multiSigPublicKey([player, opponent]);
+		const tx = new Transaction();
+
+		const game = tx.moveCall({
+			target: `${this.packageId}::owned::new`,
+			arguments: [
+				tx.pure.address(player.toSuiAddress()),
+				tx.pure.address(opponent.toSuiAddress()),
+				tx.pure(bcs.vector(bcs.u8()).serialize(admin.toRawBytes()).toBytes()),
```

Comment:
I think what you did is the recommended way. @hayes-mysten to confirm!

Reply:
This is mostly the correct way to do this, but the last toBytes isn't needed: `tx.pure(bcs.vector(bcs.u8()).serialize(admin.toRawBytes()))`



`bcs.bytes()` is an existing method that is used for fixedLength bytes, so I think also calling vector<u8> bytes would be confusing.

I just opened a PR to make this a little better though: https://github.com/MystenLabs/sui/pull/18561/files

This will add `tx.pure.vector('u8', admin.toRawBytes())`